pinot-spi/src/main/java/org/apache/pinot/spi/env/SegmentMetadataPropertyReader.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.env;
+
+import com.google.common.base.Preconditions;
+import java.io.Reader;
+import org.apache.commons.configuration2.PropertiesConfiguration.PropertiesReader;
+
+
+/**
+ * SegmentMetadataPropertyReader extends the PropertiesReader
+ * <p>
+ * Purpose: loads the segment metadata faster
+ *  - by skipping the unescaping of key and
+ *  - parsing the line by splitting based on first occurrence of separator
+ */
+class SegmentMetadataPropertyReader extends PropertiesReader {
+
+  public SegmentMetadataPropertyReader(Reader reader) {
+    super(reader);
+  }
+
+  @Override
+  protected void parseProperty(final String line) {
+    // skip the regex based parsing of the line content and splitting the content based on first occurrence of separator
+    String[] keyValue = line.split(getPropertySeparator());
+    Preconditions.checkArgument(keyValue.length == 2, "property content split should result in key and value");
+    initPropertyName(keyValue[0]);
+    initPropertyValue(keyValue[1]);
+    initPropertySeparator(getPropertySeparator());
```

Comment:
Good point. 

I think it only happens if we change the seprator configuration while writing the configuration file. By default, it's `=` and I don't think we want to change that. I can add a specific comment about it for future reference and maybe the unit test around it. Thanks

Reply:
currently, how we change the separator configuration while writing the configuration file? Is there a way to force that to be '=' as part of contract of using version 2.

Reply:
We are not changing the separator configuration. However, we only have [write method ](https://github.com/apache/pinot/pull/12440/files#diff-d500a9bb6a9a2455521f13b93599aeb1db76a0a3a285ffb6a4cef58dcf98e9c7R189), and we can explicitly set the separator value so that in future if default behaviour of commons-configuration2 changes, it wouldn't impact us.

Reply:
Setting the global separator for versioned properties configurations ensures it gets written with the `=` separator. Also, I added the unit test case. Thanks